settings.gradle.kts

```diff
@@ -60,3 +60,36 @@ gradleEnterprise {
         publishAlwaysIf(isCiBuild)
     }
 }
+
+@Suppress("UnstableApiUsage") // Central declaration of repositories is an incubating feature
+dependencyResolutionManagement {
+
+    repositoriesMode.set(RepositoriesMode.PREFER_SETTINGS)
```

Comment:
I actually wonder if the default `PREFER_PROJECT` would be better

It looks like we have a few projects (namely, `intellij-dependency` and `frontend`) that need very specific repositories (like the for the gradle-node plugin) that are unlikely to be used in any other project. Similar to dependencies, having very specific repositories be declared in that project locally would make it somewhat easier to reason about

Having truly common repositories (i.e used in >1 module) here totally makes sense though, so no questions here

wdyt?

Reply:
How about `FAIL_ON_PROJECT_REPOS`?

Reply:
`PREFER_PROJECT` is the default behaviour, so I'll set it to be explicit, and I've added a comment to explain.

I've added in some config to tell Gradle that the Yarn/Node repos can only provide Yarn/Node - so Gradle can avoid using those repositories for other dependencies.